odoo/addons/test_http/tests/test_http.py

```diff
@@ -1,20 +1,28 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 import json
+import unittest
+from ipaddress import ip_address
+from os import getenv
+from socket import socket, gethostbyname
 from unittest.mock import patch
 from urllib.parse import urlparse
-from socket import gethostbyname
 
 import odoo
 from odoo.http import Request, Session
 from odoo.tests import tagged
 from odoo.tests.common import HOST, HttpCase, new_test_user
-from odoo.tools import config, file_open, mute_logger
+from odoo.tools import config, file_open, file_path, mute_logger
 from odoo.tools.func import lazy_property
+from odoo.tools.osutil import is_host_up
 from odoo.addons.test_http.controllers import CT_JSON
 from odoo.addons.test_http.utils import MemorySessionStore, HtmlTokenizer
 
 
+WEB_SERVER_URL = urlparse(getenv('ODOO_WEB_SERVER_URL', 'http://odoo.localhost:80'))
+SKIP_WEB_SERVER_TESTS = not is_host_up(WEB_SERVER_URL.hostname, WEB_SERVER_URL.port)
```

Comment:
@odoo/rd-runbot Hello there, I'm adding some unittests that require to request Odoo though nginx, you can see here that the tests are disabled in case there is no nginx server running (I mean... nothing on odoo.localhost:80 which is the address/port that nginx binds on my own laptop). I made it configurable via that un-documented environment variable so that you guys can enable those tests on runbot too :)

Reply:
Adding unit tests which can't actually be run doesn't seem very useful, this should be a separate unrelated file, or possibly a separate test script which is run as a standalone.

That way it can also be self-documenting with a proper help and everything.

Reply:
Still TODO, I did give a try for the executable test file but I wasn't a fan of the solution. I'll give it another go